### PR TITLE
docs: fix some vague description about analysis arguments

### PR DESCRIPTION
**Pull request by huikang**
Signed-off-by: Hui Kang <hui.kang@salesforce.com>

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: docs/features/analysis.md [hunk 1]
```diff
@@ -356,7 +356,7 @@ templates together. The controller combines the `metrics` and `args` fields of a
     The controller will error when merging the templates if:
 
     * Multiple metrics in the templates have the same name
-    * Two arguments with the same name both have values
+    * Two arguments with the same name have different default values no matter the argument value in Rollout
```
Comment (agrawroh) [inline docs/features/analysis.md:359]: I failed to understand this completely. Is this more clear by any chance?

_"Two arguments with the same name have different default values. It doesn't matter what the argument values are in the Rollout."_

File: docs/features/analysis.md [hunk 2]
```diff
@@ -370,11 +370,11 @@ metadata:
   name: args-example
 spec:
   args:
-  # required
+  # required in Rollout due to no default value
```
Comment (agrawroh) [inline docs/features/analysis.md:373]:
```suggestion
  # required in the Rollout due to no default value
```

File: docs/features/analysis.md [hunk 3]
```diff
@@ -370,11 +370,11 @@ metadata:
...
   - name: service-name
   - name: stable-hash
   - name: latest-hash
-  # optional
+  # optional in Rollout given the default value
```
Comment (agrawroh) [inline docs/features/analysis.md:377]:
```suggestion
  # optional in the Rollout given the default value
```

**Review by agrawroh — APPROVED** [review]
```
                 —— SHIP IT ——

                   ,:',:`,:'
                __||_||_||_||__
           ____["""""""""""""""]____
           \ " '''''''''''''''''''' |
    ~^~~^~^~^^~^~^~^~^~^~^~^~~^~^~^^~~^~^
```

File: docs/features/analysis.md (hunk 1 above)
Comment (huikang) [reply docs/features/analysis.md:359]: The original text is wrong in that if `Two arguments have different default values`, the merging is valid. So the updated text adds the additional condition of `have different default values`

Comment (agrawroh) [reply docs/features/analysis.md:359]: Got it. Thanks for explaining :)

**Review by jessesuen — APPROVED** [review]

**Pull request merged**